base/abstractarray.jl

```diff
@@ -208,6 +208,13 @@ reshape(a::AbstractArray, dims::Int...) = reshape(a, dims)
 
 ## from general iterable to any array
 
+function copy!(dest, src)
+    for x in src
+        push!(dest, x)
```

Comment:
In the case of `copy!`, that `f!` seems like it should be `setindex!`. The generic implementation for `dest::AbstractArray` should probably be
```jl
for (i, x) in zip(eachindex(dest), src)
    dest[i] = x
end
```